packages/smooth_app/lib/pages/personalized_ranking_page.dart

```diff
@@ -58,9 +60,41 @@ class _PersonalizedRankingPageState extends State<PersonalizedRankingPage>
         context.watch<ProductPreferences>();
     context.watch<LocalDatabase>();
     final AppLocalizations appLocalizations = AppLocalizations.of(context);
+
+    Future<void> _addToList() async {
```

Comment:
I'm really not a big fan of putting the methods anywhere as fields. Like, methods are methods. In addition to that, if it's a field it's not private, so it should be `addToList` and not `_addToList`. And btw there are several listS to add to, so  `addToLists` or `_addToLists`.

But it's a matter of personal taste (those days I've been reflecting on code in general and I think I prefer code that can be reused as is in different languages) (and that field/method thing is not possible in Java) (but that's another story).

Reply:
I don't think I can really follow you here, what would be your prefered alternative, putting the logic inline?

Reply:
Here I would have considered `_addToList()` as a private method of class `_PersonalizedRankingPageState`, not as an inner method field of method `build(BuildContext)`.